src/lsst/rsp/_discovery.py

```diff
@@ -0,0 +1,220 @@
+"""Service discovery within Nublado notebooks.
+
+These function suse a service discovery JSON file created by the Nublado
+controller when the user's lab was created for everything except obtaining
+InfluxDB credentials. This avoids a dependency on the Rubin Repertoire client,
+which may require a newer Python version than is available in the execution
+environment using lsst.rsp.
+"""
+
+from __future__ import annotations
+
+import json
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any
+
+import httpx
+
+from ._exceptions import (
+    DatasetNotSupportedError,
+    DiscoveryNotAvailableError,
+    InvalidDiscoveryError,
+    TokenNotAvailableError,
+    UnknownDatasetError,
+    UnknownInfluxDBError,
+    UnknownServiceError,
+)
+from .utils import get_access_token
+
+_DISCOVERY_PATH = Path("/etc/nublado/discovery/v1.json")
+"""Path to static service discovery information."""
+
+__all__ = [
+    "InfluxDBCredentials",
+    "InfluxDBLocation",
+    "get_influxdb_credentials",
+    "get_influxdb_location",
+    "get_service_url",
+]
+
+
+@dataclass
+class InfluxDBLocation:
+    """Location information for an InfluxDB database."""
+
+    url: str
+    """URL to the InfluxDB database."""
+
+    database: str
+    """Name of the InfluxDB database."""
+
+    schema_registry: str
+    """URL to the Schema Registry corresponding to that database."""
+
+
+@dataclass
+class InfluxDBCredentials(InfluxDBLocation):
+    """InfluxDB database location information and credentials."""
+
+    username: str
+    """Username to use for authentication."""
+
+    password: str
+    """Password to use for authentication."""
+
+
+def _get_discovery() -> dict[str, Any]:
+    """Get the data and service discovery information.
+
+    Returns
+    -------
+    dict
+        Parsed service discovery information.
+
+    Raises
+    ------
+    DiscoveryNotAvailableError
+        Raised if no service discovery information is available.
+    """
+    try:
+        with _DISCOVERY_PATH.open("r") as fh:
```

Comment:
I prefer json.loads(path.read_text()) these days, but ymmv

Reply:
Yeah, that's probably clearer here. It takes more memory than this approach, but for this tiny JSON file that doesn't really matter.